### PR TITLE
Extend docs

### PR DESCRIPTION
**Pull request by KEggensperger**

### CODE REVIEW

**Review by mfeurer — CHANGES_REQUESTED** [review]

File: autosklearn/experimental/askl2.py [hunk 1]
```diff
@@ -272,10 +278,10 @@ def __init__(
             fine-grained information on what to save. Allowed elements in the
             list are:
 
-            * ``'y_optimization'`` : do not save the predictions for the
+            * ``y_optimization`` : do not save the predictions for the
```
[the remark this answers is not in the document]
Comment (KEggensperger) [reply autosklearn/experimental/askl2.py]: done

File: doc/faq.rst [hunk 2]
```diff
@@ -6,267 +6,520 @@
...
+    For further information about how scikit-learn handles multiprocessing, please check the `Parallelism, resource management, and configuration <https://scikit-learn.org/stable/computing/parallelism.html>`_ documentation from the library.
 
-The short answer: no.
 
-The long answer answer is a bit more nuanced: maintaining Auto-sklearn requires a lot of time and
-effort, which would grow even larger when depending on more libraries. Also, adding more
-libraries would require us to generate meta-data more often. Lastly, having more choices does not
-guarantee a better performance for most users as having more choices demands a longer search for
-good models and can lead to more overfitting.
+.. collapse:: <b>Auto-sklearn is extremely memory hungry in a sequential setting</b>
```
Comment (eddiebergman) [inline doc/faq.rst:144]: @mfeurer take a look at this.

File: doc/faq.rst [hunk 3]
```diff
@@ -6,267 +6,520 @@
...
+            )
+            plt.show()
+
+.. collapse:: <b>Which models were evaluated?</b>
+
+    The results obtained from the final ensemble can be printed by calling ``show_models()``.
+
+.. collapse:: <b>Which models are in the final ensemble?</b>
+
+    ``automl.leaderboard()`` shows the ensemble members.
```
Comment (eddiebergman) [inline doc/faq.rst:254]: `leaderboard()` can show all evaluated models.

```rst
.. collapse:: <b>Which models were evaluated?</b>

    You can see all models evaluated using :meth:`automl.leaderboard(ensemble_only=False) <autosklearn.classification.AutoSklearnClassifier.leaderboard>`.
    
.. collapse:: <b>Which models are in the final ensemble?</b>

    Use either :meth:`automl.leaderboard(ensemble_only=True) <autosklearn.classification.AutoSklearnClassifier.leaderboard>` or ``automl.show_models()``
```

File: doc/faq.rst [hunk 4]
```diff
@@ -6,267 +6,520 @@
...
+
+.. collapse:: <b>Is there more data I can look at?</b>
+
+    ``cv_results_`` returns a dict with keys as column headers and values as columns, that can be imported into
+    a pandas DataFrame.
```
Comment (eddiebergman) [inline doc/faq.rst:259]:
```suggestion
    a pandas DataFrame, e.g. ``df = pd.DataFrame(automl.cv_results_)``
```

File: doc/faq.rst [hunk 5]
```diff
@@ -6,267 +6,520 @@
...
+
+.. collapse:: <b>Where does Auto-sklearn output files by default?</b>
+
+    *Auto-sklearn* heavily uses the hard drive to store temporary data, models and log files which can
+    be used to inspect the behavior of Auto-sklearn. Each run of Auto-sklearn requires
+    its own directory. If not provided by the user, *Auto-sklearn* requests a temporary directory from
+    Python, which by default is located under ``/tmp`` and starts with ``autosklearn_tmp_`` followed
+    by a random string. By default, this directory is deleted when the *Auto-sklearn* object is
+    destroyed. If you want to keep these files you can pass the argument
```
Comment (eddiebergman) [inline doc/faq.rst:268]:
```suggestion
    finished fitting. If you want to keep these files you can pass the argument
```

File: doc/faq.rst [hunk 6]
```diff
@@ -6,267 +6,520 @@
...
+    ``delete_tmp_folder_after_terminate=True`` to the *Auto-sklearn* object.
+
+    The :class:`autosklearn.classification.AutoSklearnClassifier` and all other *auto-sklearn*
+    estimators accept the argument ``tmp_directory`` which change where such output is written to.
```
Comment (eddiebergman) [inline doc/faq.rst:272]:
```suggestion
    estimators accept the argument ``tmp_folder`` which change where such output is written to.
```

File: doc/faq.rst [hunk 7]
```diff
@@ -6,267 +6,520 @@
...
+
+    There's an additional argument ``output_directory`` which can be passed to *Auto-sklearn* and it
+    controls where test predictions of the ensemble are stored if the test set is passed to ``fit()``.
```
Comment (eddiebergman) [inline doc/faq.rst:275]: We don't have this section, needs to be deleted

File: doc/faq.rst [hunk 8]
```diff
@@ -6,267 +6,520 @@
...
+    are only removed from disk when the ensemble builder finishes an iteration.
+
+    .. note::
+
+        Especially when running in parallel it can happen that multiple models are constructed during
+        one run of the ensemble builder and thus *Auto-sklearn* can exceed the given limit.
+
+    .. note::
+
+       These limits do only apply to models and their predictions, but not to other files stored in
```
Comment (eddiebergman) [inline doc/faq.rst:303]:
```suggestion
       These limits only apply to models and their predictions, not to other files stored in
```

File: doc/faq.rst [hunk 9]
```diff
@@ -6,267 +6,520 @@
...
-Only use interpretable models
------------------------------
+.. collapse:: <b>How can I restrict the searchspace?</b>
 
-Auto-sklearn can be restricted to only use interpretable models and preprocessing algorithms.
-Please see the Section :ref:`restricting_the_searchspace` to learn how to restrict the models
-which are searched over or see the Example
-:ref:`sphx_glr_examples_40_advanced_example_interpretable_models.py`.
+ The following shows an example of how to exclude all preprocessing methods and restrict the configuration space to
+ only random forests.
```
Comment (eddiebergman) [inline doc/faq.rst:312]: These 2 lines need to be indented I guess

File: doc/faq.rst [hunk 10]
```diff
@@ -6,267 +6,520 @@
...
+        predictions = automl.predict(X_test)
 
-In certain cases, for example for debugging, it can be helpful to limit the number of
-model evaluations. We do not provide this as an argument in the API as we believe that it
-should NOT be used in practice, but that the user should rather provide time limits.
-An example on how to add the number of models to try as an additional stopping condition
-can be found `in this github issue <https://github.com/automl/auto-sklearn/issues/451#issuecomment-376445607>`_.
-Please note that Auto-sklearn will stop when either the time limit or the number of
-models termination condition is reached.
+    **Note:** The strings used to identify estimators and preprocessors are the filenames without *.py*.
```
Comment (eddiebergman) [inline doc/faq.rst:327]:
```suggestion
    **Note:** The strings used to identify estimators and preprocessors are the filenames in `components <https://github.com/automl/auto-sklearn/tree/master/autosklearn/pipeline/components>`_ without *.py*.
```

File: doc/faq.rst [hunk 11]
```diff
@@ -6,267 +6,520 @@
...
+    The long answer answer is a bit more nuanced: maintaining Auto-sklearn requires a lot of time and
+    effort, which would grow even larger when depending on more libraries. Also, adding more
+    libraries would require us to generate meta-data more often. Lastly, having more choices does not
+    guarantee a better performance for most users as having more choices demands a longer search for
+    good models and can lead to more overfitting.
+
+    Nevertheless, everyone can still add their favorite model to Auto-sklearn's search space by
+    following the `examples on how to extend Auto-sklearn
+    <https://automl.github.io/auto-sklearn/master/examples/index.html#extension-examples>`_.
+
```
Comment (eddiebergman) [inline doc/faq.rst:369]: I'm not sure distributing this over two lines will work, would need to check the generated docs.

File: doc/faq.rst [hunk 12]
```diff
@@ -6,267 +6,520 @@
...
+.. collapse:: <b>How can I only search for interpretable models</b>
+
+    Auto-sklearn can be restricted to only use interpretable models and preprocessing algorithms.
+    Please see the Section :ref:`space` to learn how to restrict the models
+    which are searched over or see the Example
+    :ref:`sphx_glr_examples_40_advanced_example_interpretable_models.py`.
+
+    We don't provide a judgement which of the models are interpretable as this is very much up to the
+    specific use case, but would like to note that decision trees and linear models usually most
```
Comment (eddiebergman) [inline doc/faq.rst:381]:
```suggestion
    specific use case, but we would like to note that decision trees and linear models are usually most
```

File: doc/faq.rst [hunk 13]
```diff
@@ -6,267 +6,520 @@
...
+    interpretable.
+
+Ensembling
+==========
+
+.. collapse:: <b>What can I configure wrt the ensemble building process?</b>
```
Comment (eddiebergman) [inline doc/faq.rst:387]:
```suggestion
.. collapse:: <b>How can I configure the ensemble building process?</b>
```

File: doc/faq.rst [hunk 14]
```diff
@@ -6,267 +6,520 @@
...
+
+.. collapse:: <b>Which models are in the final ensemble?</b>
+
+    The results obtained from the final ensemble can be printed by calling ``show_models()`` or  ``leaderboard()``.
+    The *auto-sklearn* ensemble is composed of scikit-learn models that can be inspected as exemplified
+    in the Example :ref:`sphx_glr_examples_40_advanced_example_get_pipeline_components.py`.
+
+.. collapse:: <b>Can I fit an ensemble also only post-hoc?</b>
+
+    To use a single core only, it is possible to build ensembles post-hoc. An example on how to do this (first searching
```
Comment (eddiebergman) [inline doc/faq.rst:403]: Not sure why the comment on single core is neccessary?

File: doc/faq.rst [hunk 15]
```diff
@@ -6,267 +6,520 @@
...
+    * ``max_models_on_disc`` defines the maximum number of models that are kept on the disc, as a mechanism to control the amount of disc space consumed by *auto-sklearn*. Throughout the automl process, different individual models are optimized, and their predictions (and other metadata) is stored on disc. The user can set the upper bound on how many models are acceptable to keep on disc, yet this variable takes priority in the definition of the number of models used by the ensemble builder (that is, the minimum of ``ensemble_size``, ``ensemble_nbest`` and ``max_models_on_disc`` determines the maximal amount of models used in the ensemble). If set to None, this feature is disabled.
+
+.. collapse:: <b>Which models are in the final ensemble?</b>
+
+    The results obtained from the final ensemble can be printed by calling ``show_models()`` or  ``leaderboard()``.
+    The *auto-sklearn* ensemble is composed of scikit-learn models that can be inspected as exemplified
+    in the Example :ref:`sphx_glr_examples_40_advanced_example_get_pipeline_components.py`.
+
+.. collapse:: <b>Can I fit an ensemble also only post-hoc?</b>
```
Comment (eddiebergman) [inline doc/faq.rst:401]:
```suggestion
.. collapse:: <b>Can I explicitly fit an ensemble post fitting?</b>
```

File: doc/faq.rst [hunk 16]
```diff
@@ -6,267 +6,520 @@
...
+
+
+Issues and Debugging
+====================
 
-For *Auto-sklearn 2.0* it is not possible to do so because of the method used to construct the
-meta-data.
+.. collapse:: <b>How can I limit the number of model evaluations for debugging?</b>
+
+    In certain cases, for example for debugging, it can be helpful to limit the number of
```
Comment (eddiebergman) [inline doc/faq.rst:462]:
```suggestion
    In certain cases such as debugging, it can be helpful to limit the number of
```

File: doc/faq.rst [hunk 17]
```diff
@@ -6,267 +6,520 @@
...
+    An example on how to add the number of models to try as an additional stopping condition
+    can be found `in this github issue <https://github.com/automl/auto-sklearn/issues/451#issuecomment-376445607>`_.
+    Please note that Auto-sklearn will stop when either the time limit or the number of
+    models termination condition is reached.
+
+.. collapse:: <b>Why does the final ensemble contains only a dummy model?</b>
+
+    This is a symptom of the problem that all runs started by Auto-sklearn failed. Usually, the issue
+    is that the runtime or memory limit were too tight. Please check the output of
+    ``sprint_statistics`` to see the distribution of why runs failed. If there are mostly crashed
```
Comment (eddiebergman) [inline doc/faq.rst:474]:
```suggestion
    ``sprint_statistics()`` to see the distribution of why runs failed. If there are mostly crashed
```

File: doc/faq.rst [hunk 18]
```diff
@@ -6,267 +6,520 @@
...
+    from scikit-learn.
 
-How is the meta-data generated?
--------------------------------
+.. collapse:: <b>Vanilla auto-sklearn</b>
 
-Auto-sklearn 1.0
-~~~~~~~~~~~~~~~~
+    In order to obtain *vanilla auto-sklearn* as used in `Efficient and Robust Automated Machine Learning
+    <https://papers.nips.cc/paper/5872-efficient-and-robust-automated-machine -learning>`_
```
Comment (eddiebergman) [inline doc/faq.rst:511]:
```suggestion
    `<https://papers.nips.cc/paper/5872-efficient-and-robust-automated-machine-learning>`_
```

File: doc/manual.rst [hunk 19]
```diff
@@ -6,232 +6,299 @@
 Manual
 ======
 
-This manual shows how to use several aspects of auto-sklearn. It either
-references the examples where possible or explains certain configurations.
+This manual give an overview of different aspects of *auto-sklearn*. For each section, we either references examples or
```
Comment (eddiebergman) [inline doc/manual.rst:9]:
```suggestion
This manual gives an overview of different aspects of *auto-sklearn*. For each section, we either references examples or
```

File: doc/manual.rst [hunk 20]
```diff
@@ -6,232 +6,299 @@
...
+
+.. collapse:: <b>Inspect the final ensemble</b>
+
+    The results obtained from the final ensemble can be printed by calling ``show_models()``.
+    The *auto-sklearn* ensemble is composed of scikit-learn models that can be inspected as exemplified
+    in the Example :ref:`sphx_glr_examples_40_advanced_example_get_pipeline_components.py`.
+
+.. collapse:: <b>Fit ensemble post-hoc</b>
+
+    To use a single core only, it is possible to build ensembles post-hoc. An example on how to do this (first searching
```
Comment (eddiebergman) [inline doc/manual.rst:143]: Same thing, not sure why the single core here is relevant to ensemble building.

File: doc/manual.rst [hunk 21]
```diff
@@ -6,232 +6,299 @@
...
+            )
+            plt.show()
+
+.. collapse:: <b>Evaluated models</b>
+
+    The results obtained from the final ensemble can be printed by calling ``show_models()``.
+
+.. collapse:: <b>Leaderboard</b>
+
+    ``automl.leaderboard()`` shows the ensemble members.
```
Comment (eddiebergman) [inline doc/manual.rst:191]:
```suggestion
    ``automl.leaderboard()`` shows the ensemble members, check the :meth:`docs <autosklearn.classification.AutoSklearnClassifier.leaderboard>` for using leaderboard for getting information on *all* runs.
```

File: doc/faq.rst (hunk 3 above)
Comment (KEggensperger) [reply doc/faq.rst]: Thanks, fixed

File: doc/manual.rst (hunk 20 above)
Comment (KEggensperger) [reply doc/manual.rst:143]: Agreed, this is not relevant (and also outdated). Fixed.

File: doc/faq.rst [hunk 22]
```diff
@@ -6,267 +6,518 @@
...
+    from scikit-learn.
 
-How is the meta-data generated?
--------------------------------
+.. collapse:: <b>Vanilla auto-sklearn</b>
 
-Auto-sklearn 1.0
-~~~~~~~~~~~~~~~~
+    In order to obtain *vanilla auto-sklearn* as used in `Efficient and Robust Automated Machine Learning
+    `<https://papers.nips.cc/paper/5872-efficient-and-robust-automated-machine-learning>`_
```
Comment (eddiebergman) [inline doc/faq.rst:509]: This fails with the doc tests, I assume it just a case of removing the backtick before the `<https ...>` and perhaps making sure it's not spread out over a linebreak.

**Review by eddiebergman — CHANGES_REQUESTED** [review]

**Pull request merged**